core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -377,6 +380,71 @@ public void addSupportedLanguage(Locale locale) {
         getLocales().loadMissingBundles();
     }
 
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalArgumentException when instance isn't an instance of clazz
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public void registerDependency(Class<?> clazz, Object instance){
+        registerNamedDependency(clazz, clazz.getName(), instance);
+    }
+
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param key the key which needs to be present if that
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalArgumentException when instance isn't an instance of clazz
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public void registerNamedDependency(Class<?> clazz, String key, Object instance){
+        if(!clazz.isInstance(instance)){
+            throw new IllegalArgumentException(instance.getClass().getName() + " isn't a subclass of " + clazz.getName() + "!");
+        }
+        if(dependencies.containsRow(clazz) && dependencies.containsColumn(key)){
+            throw new IllegalStateException("There is already an instance of " + clazz.getName() + " with the key " + key + " registered!");
```

Comment:
Not sure we should throw, but just let it overwrite, maybe return old, like a typical put operation.